bpf/straceback-guess-bpf.c

```diff
@@ -233,8 +233,13 @@ static u32 find_prog_idx(void *ctx, u64 pid, u32 utsns, long id, int exit) {
 			// Program already installed. Use it.
 			return status->idx;
 		} else if (status->status == CONTAINER_STATUS_WAITING) {
+			// If runc calls mount() with a path starting with:
+			// /sys/fs/cgroup/systemd/
+			// then send that full path to the traceloop process
+			// so it can extract the PodUID and ContainerID from
+			// the path.
 			if (exit != 0 && id != 165) { // mount
-				return 0xFFFFFFFF;
+				return status->idx;
```

Comment:
I don't understand these changes - whatever happens, we always return `status->idx`.

Reply:
That's correct. The difference is whether we call `bpf_perf_event_output()` below or not to inform the traceloop process.